lasy-fuzzy-syntax.md

```diff
@@ -0,0 +1,25 @@
+## motivation
+Описывает набор инструкций, которые будут выполняться только тогда, когда внешнему коду потребуется результат исполнения этих инструкций.
+
+## примеры из других языков
+LINQ в .NET
+
+## high-level api
+```javascript
+const result = lasy array.map(...).filter(...).reduce(...);
+// инструкции не будут вычисляться вообще
+...
+
+console.log(result[0]) // result будет вычислен только сейчас.
+```
+
+### Принудительное исполнение
+```javascript
+const result = lasy array.map(...).filter(...);
+// result не будет вычислен вообще
+...
+(fuzzy result).reduce(...);
+// вычисляется значение на момент вызова fuzzy, а затем в очередь вычислений добавляется новый вызов
```

Comment:
the value at the time of calling fuzzy is calculated, and then a new call is added to the calculation queue